app/src/sharedTest/java/org/oppia/android/app/onboarding/CreateProfileFragmentTest.kt

```diff
@@ -451,6 +466,104 @@ class CreateProfileFragmentTest {
     }
   }
 
+  @Test
+  fun testFragment_inputNameWithNumbers_showsNameOnlyLettersError() {
+    launchNewLearnerProfileActivity().use {
+      onView(withId(R.id.create_profile_nickname_edittext))
+        .perform(
+          editTextInputAction.appendText("John123"),
+          closeSoftKeyboard()
+        )
+
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.onboarding_navigation_continue)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.create_profile_nickname_error))
+        .check(matches(withText(R.string.add_profile_error_name_only_letters)))
+    }
+  }
+
+  @Test
+  fun testFragment_configChange_inputNameWithNumbers_showsNameOnlyLettersError() {
+    launchNewLearnerProfileActivity().use {
+      onView(isRoot()).perform(orientationLandscape())
+
+      onView(withId(R.id.create_profile_nickname_edittext))
+        .perform(
+          editTextInputAction.appendText("John123"),
+          closeSoftKeyboard()
+        )
+
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.onboarding_navigation_continue)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.create_profile_nickname_error))
+        .check(matches(withText(R.string.add_profile_error_name_only_letters)))
+    }
+  }
+
+  @Test
+  fun testFragment_inputNameWithNumbers_thenInputNameWithLetters_errorIsCleared() {
+    launchNewLearnerProfileActivity().use {
+      onView(withId(R.id.create_profile_nickname_edittext))
+        .perform(
+          editTextInputAction.appendText("John123"),
+          closeSoftKeyboard()
+        )
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.onboarding_navigation_continue)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.create_profile_nickname_error))
+        .check(matches(withText(R.string.add_profile_error_name_only_letters)))
+
+      onView(withId(R.id.create_profile_nickname_edittext))
+        .perform(
+          editTextInputAction.appendText("John"),
+          closeSoftKeyboard()
+        )
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.create_profile_nickname_error))
+        .check(matches(withEffectiveVisibility(Visibility.GONE)))
+    }
+  }
+
+  @Test
+  fun testFragment_configChange_inputNameWithNumbers_thenInputNameWithLetters_errorIsCleared() {
```

Comment:
For here & the above: these are technically testing landscape, not configuration changes. I suggest renaming them, and adding separate tests for verifying that a configuration change after the error is initiated retains the error (for both portrait & landscape). That's the more "interesting" behavior to test for config changes.

Reply:
Updated as suggested.